src/modrover/rover.py

```diff
@@ -46,10 +51,26 @@ def performances(self):
         """Convenience property to select only the performance of each learner."""
         return {lid: learner.performance for lid, learner in self.learners.items()}
 
-    def get_learner(self, learner_id: LearnerID) -> Learner:
+    @property
+    def num_covariates(self):
+        # TODO: Validate this method, check for overlaps and such
+        num_covariates = 0
+        for key, val in self.col_fixed.items():
+            num_covariates += len(val)
+
+        for key, val in self.col_explore.items():
+            num_covariates += len(val)
+
+        return num_covariates
+
+    def check_is_fitted(self):
+        if not self.super_learner:
+            raise NotFittedError("Rover has not been ensembled yet")
+
+    def get_learner(self, learner_id: LearnerID, use_cache: bool = True) -> Learner:
 
         # See if we've already initialized one
-        if learner_id in self.learners:
+        if learner_id in self.learners and use_cache:
             return self.learners[learner_id]
 
         all_covariates = list(self.col_explore.values())[0]
```

Comment:
as a general note, I had some trouble with indexing in implementing this PR - had a lot of off-by-one type errors